web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateSearchApiController.java

```diff
@@ -76,4 +75,207 @@
         }
         return new ResponseEntity<>(treatments, status);
     }
+
+    @Override
+    public ResponseEntity<LinkedHashSet<TypeaheadSearchResp>> searchTypeAheadGet(
+        @ApiParam(value = "The search query, it could be hugoSymbol, entrezGeneId or variant.", required = true) @RequestParam(value = "query") String query,
+        @ApiParam(value = "The limit of returned result.") @RequestParam(value = "limit", defaultValue = "5", required = false) Integer limit) {
+        LinkedHashSet<TypeaheadSearchResp> result = new LinkedHashSet<>();
+        if (limit == null) {
+            limit = TYPEAHEAD_RETURN_LIMIT;
+        }
+        if (query != null) {
+            List<String> keywords = Arrays.asList(query.trim().split("\\s+"));
+
+            if (keywords.size() == 1) {
+                // Blur search gene
+                result.addAll(convertGene(GeneUtils.searchGene(keywords.get(0), false), keywords.get(0)));
+
+                // Blur search variant
+                result.addAll(convertVariant(AlterationUtils.lookupVariant(keywords.get(0), false, AlterationUtils.getAllAlterations()), keywords.get(0)));
+            } else if (keywords.size() == 2) {
+                // Assume one of the keyword is gene
+                Map<String, Set<Gene>> map = new HashedMap();
+                for (String keyword : keywords) {
+                    map.put(keyword, GeneUtils.searchGene(keyword, true));
+                }
+
+                //Find exact match
+                result.addAll(getMatch(map, keywords, true));
+                result.addAll(getMatch(map, keywords, false));
```

Comment:
why do we need to getMatch twice? Doesn't not exact match result include exact match result?